src/app/problems/problem-create/java-create-snippet/java-create-snippet.component.ts

```diff
@@ -55,28 +56,47 @@ export class JavaCreateSnippetComponent implements OnInit {
     }
 
     /**
-     * Select a course from the given event.
-     * @param value - The event.
+     * Select a course from the given DOM event.
+     * @param value - The DOM event.
      */
-    courseSelectedEvent(value: Event): void {
-        this.courseSelectedById(+(value.target as HTMLInputElement).value);
+    courseSelectionChanged(value: Event): void {
```

Comment:
Are you sure about the type `Event` here?

Reply:
Yeah. I could just pass the value directly to the other functions instead in the `(change)` instead of passing a DOM event?

Reply:
Let's keep it this way! I'm not sure which is better tho.